esphome/components/pmsx003/pmsx003.cpp

```diff
@@ -116,55 +129,44 @@ void PMSX003Component::loop() {
     }
   }
 }
+
 float PMSX003Component::get_setup_priority() const { return setup_priority::DATA; }
+
 optional<bool> PMSX003Component::check_byte_() {
-  uint8_t index = this->data_index_;
-  uint8_t byte = this->data_[index];
+  const uint8_t index = this->data_index_;
+  const uint8_t byte = this->data_[index];
 
   if (index == 0)
-    return byte == 0x42;
+    return byte == START_CHARACTER_1;
 
   if (index == 1)
-    return byte == 0x4D;
+    return byte == START_CHARACTER_2;
 
   if (index == 2)
     return true;
 
-  uint16_t payload_length = this->get_16_bit_uint_(2);
+  const uint16_t payload_length = this->get_16_bit_uint_(2);
   if (index == 3) {
-    bool length_matches = false;
-    switch (this->type_) {
-      case PMSX003_TYPE_X003:
-        length_matches = payload_length == 28 || payload_length == 20;
-        break;
-      case PMSX003_TYPE_5003T:
-      case PMSX003_TYPE_5003S:
-        length_matches = payload_length == 28;
-        break;
-      case PMSX003_TYPE_5003ST:
-        length_matches = payload_length == 36;
-        break;
-    }
-
-    if (!length_matches) {
+    if (!this->check_payload_length_(payload_length)) {
       ESP_LOGW(TAG, "PMSX003 length %u doesn't match. Are you using the correct PMSX003 type?", payload_length);
       return false;
     }
     return true;
```

Comment:
Could this if (index == *) be reduced to switch (index) ?

Reply:
i think easily only for 0-2. as then other code gets executed before the next check. but then we have some check with the switch and some with if. i would prefer to keep the if's as then it is consistent. or do you describe more in detail what you want

Reply:
@DjordjeMandic anything that i should do here now? if yes pls a bit more details

Reply:
```suggestion

  switch (index) {
    case 0:
      return byte == START_CHARACTER_1;
    case 1:
      return byte == START_CHARACTER_2;
    case 2:
      return true;
    case 3:
      if (!this->check_payload_length_(payload_length)) {
        ESP_LOGW(TAG, "PMSX003 length %u doesn't match. Are you using the correct PMSX003 type?", payload_length);
        return false;
      }
      return true;
  }

```

Maybe like this? You will also need to move `const uint16_t payload_length = this->get_16_bit_uint_(2);` before switch statement. For example:
```cpp
optional<bool> PMSX003Component::check_byte_() {
  const uint8_t index = this->data_index_;
  const uint8_t byte = this->data_[index];
  const uint16_t payload_length = this->get_16_bit_uint_(2);

  switch (index) {
    case 0:
      return byte == START_CHARACTER_1;
    case 1:
      return byte == START_CHARACTER_2;
    case 2:
      return true;
    case 3:
      if (!this->check_payload_length_(payload_length)) {
        ESP_LOGW(TAG, "PMSX003 length %u doesn't match. Are you using the correct PMSX003 type?", payload_length);
        return false;
      }
      return true;
  }
```

Reply:
and the rest of the function code into the "default" of the switch or after the switch?
i personally like the if (only) variant more

Reply:
> and the rest of the function code into the "default" of the switch or after the switch?
> i personally like the if (only) variant more

Rest after the switch but if you like it how it is now then it's fine.